src/ngMock/angular-mocks.js

```diff
@@ -193,6 +239,72 @@ angular.mock.$Browser.prototype = {
   }
 };
 
+/**
+ * @ngdoc function
+ * @name $flushPendingTasks
+ *
+ * @description
+ * Flushes all pending tasks and executes the corresponding callbacks.
+ *
+ * The types of tasks that are flushed include:
+ *
+ * - Pending timeouts (via {@link $timeout}).
+ * - Pending tasks scheduled via {@link ng.$rootScope.Scope#$applyAsync}.
+ * - Pending tasks scheduled via {@link ng.$rootScope.Scope#$evalAsync}.
+ *   These include tasks scheduled via `$evalAsync()` indirectly (such as {@link $q} promises).
+ *
+ * <div class="alert alert-info">
+ *   Periodic tasks scheduled via {@link $interval} use a different queue and are flushed by
+ *   `$flushPendingTasks()`. Use {@link ngMock.$interval#flush $interval.flush([millis])} instead.
+ * </div>
+ *
+ * @param {number=} delay - The number of milliseconds to flush.
```

Comment:
What do you mean? It doesn't apply to `$timeout` only (I ahve listed the types of tasks flushed above).

Reply:
I mean "the number milliseconds to flush" only applies to the $timeout queue. This might have been obvious when flush() was on $timeout, but now the flush() also flushes applyAsync and evalAsync and they don't have a "timeline" like timeout does. At least I would expect if I put in a delay of 0 or no delay that applyAsync and evalAsync would still be flushed, but timeouts wouldn't

Reply:
Good point. I'll mention that. I will ensure there are tests that verify the expected behavior (no delay flushes everything, 0 delay flushes stuff with 0 delay only, etc.).